controllers/designatemdns_controller.go

```diff
@@ -641,45 +627,41 @@ func (r *DesignateMdnsReconciler) reconcileNormal(ctx context.Context, instance
 			return ctrl.Result{RequeueAfter: time.Duration(1) * time.Second}, nil
 		}
 
-		if instance.Status.ReadyCount == instance.Status.DesiredNumberScheduled {
+		if instance.Status.ReadyCount == instance.Spec.Replicas {
```

Comment:
should be
` if instance.Status.ReadyCount == *instance.Spec.Replicas {`

Reply:
Thanks, I have just fixed it.